backend/src/data/repositories/group/group.repository.ts

```diff
@@ -45,6 +45,10 @@ class Group {
 
     return group ?? null;
   }
+
+  async delete(groupId: number): Promise<number> {
+    return this.#GroupModel.query().delete().where({ id: groupId });
+  }
```

Comment:
```suggestion
  delete(groupId: number): Promise<number> {
    return this.#GroupModel.query().delete().where({ id: groupId }).execute();
  }
```
as an option

Reply:
As far as I understood the only difference it would make is that it will return a promise that resolves to query result. Can you give me a hint why we would want that here?

Reply:
Usually we use `async` only when we want to use `await` inside.  
In this case Objectionjs returns a QueryBuilder instance and `async` needs to forcefully convert the return type to Promise. We can mutate builder outside of the repository (which is not good).
With `.execute()` we implicitly tell objection that we won't change query anymore (and this returns Promise)